### ISSUE TITLE
Restarting restartable.py

### ISSUE THREAD

**Issue by dbrgn** [saltyrtc/saltyrtc-server-python#15]
1. Start server: `python examples/restartable.py`
2. Send SIGHUP: `kill -HUP <pid>`. Nothing happens.
3. Send signal again. This stack trace is the result:

```
Traceback (most recent call last):
  File "examples/restartable.py", line 79, in <module>
    main()
  File "examples/restartable.py", line 53, in main
    loop.run_until_complete(restart_signal)
  File "/usr/lib64/python3.5/asyncio/base_events.py", line 375, in run_until_complete
    self.run_forever()
  File "/usr/lib64/python3.5/asyncio/base_events.py", line 345, in run_forever
    self._run_once()
  File "/usr/lib64/python3.5/asyncio/base_events.py", line 1257, in _run_once
    event_list = self._selector.select(timeout)
  File "/usr/lib64/python3.5/selectors.py", line 441, in select
    fd_event_list = self._epoll.poll(timeout, max_ev)
  File "examples/restartable.py", line 46, in restart_signal_handler
    restart_signal.set_result(True)
  File "/usr/lib64/python3.5/asyncio/futures.py", line 329, in set_result
    raise InvalidStateError('{}: {!r}'.format(self._state, self))
asyncio.futures.InvalidStateError: FINISHED: <Future finished result=True created at examples/restartable.py:43>
```

**Comment by lgrahl** [comment]
Really weird. Seems like the function `restart_signal_handler` is not being re-created in your Python version.

**Comment by lgrahl** [comment]
Just tested it. It's an issue with Python 3.5. This doesn't seem right.

**Comment by lgrahl** [comment]
Seemed to be a threading issue. Although I was pretty sure that signals are being called from the same thread the main loop runs on. Anyway, fixed in e305b5c7dc50d0ef6cd614490c05e9f029196b51
